### PR TITLE
[feature](jsonb type)refactor JSONB type using column and add testcase

### PR DESCRIPTION
**Pull request by xiaokang**
# Proposed changes

Issue Number: close [DSIP-016: Support JSON type](https://cwiki.apache.org/confluence/display/DORIS/DSIP-016%3A+Support+JSON+type?src=contextnavpagetreemode)

## Problem summary

1. Refactor JSONB type using ColumnString instead making a copy.
2. Add regression testcase for JSONB load and functions.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
3. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
4. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
5. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
6. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 38.42 seconds
 load time: 560 seconds
 storage size: 17154644069 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221030181308_clickbench_pr_35879.html

File: be/src/vec/sink/vmysql_result_writer.cpp [hunk 1]
```diff
@@ -110,14 +110,8 @@ Status VMysqlResultWriter::_add_one_column(const ColumnPtr& column_ptr,
...
                 const auto json_val = column->get_data_at(i);
-                if (json_val.data == nullptr) {
-                    if (json_val.size == 0) {
-                        // 0x01 is a magic num, not useful actually, just for present ""
-                        char* tmp_val = reinterpret_cast<char*>(0x01);
-                        buf_ret = _buffer.push_string(tmp_val, json_val.size);
-                    } else {
-                        buf_ret = _buffer.push_null();
-                    }
+                if (json_val.data == nullptr || json_val.size == 0) {
```
Comment (morningman) [inline be/src/vec/sink/vmysql_result_writer.cpp:113]: Do we need to distinguish "null" and "empty string"?

File: fe/fe-core/src/main/jflex/sql_scanner.flex [hunk 2]
```diff
@@ -270,7 +270,7 @@ import org.apache.doris.qe.SqlModeHelper;
         keywordMap.put("isolation", new Integer(SqlParserSymbols.KW_ISOLATION));
         keywordMap.put("job", new Integer(SqlParserSymbols.KW_JOB));
         keywordMap.put("join", new Integer(SqlParserSymbols.KW_JOIN));
-        keywordMap.put("json", new Integer(SqlParserSymbols.KW_JSON));
+        keywordMap.put("jsonb", new Integer(SqlParserSymbols.KW_JSONB));
```
Comment (morningman) [inline fe/fe-core/src/main/jflex/sql_scanner.flex:273]: why changing `json` to `jsonb`?
I think `jsonb` is an internal implementation, so for user interface, it is better be "json"?

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**